### PR TITLE
Document FLP DaemonSet / Deployment kinds subtelties

### PR DESCRIPTION
**Pull request by jotak**

### CODE REVIEW

File: api/v1alpha1/flowcollector_types.go [hunk 1]
```diff
@@ -192,7 +192,10 @@ type FlowCollectorFLP struct {
 
 	//+kubebuilder:validation:Enum=DaemonSet;Deployment
 	//+kubebuilder:default:=DaemonSet
-	// Kind is the workload kind, either DaemonSet or Deployment
+	// Kind is the workload kind, either DaemonSet or Deployment. When DaemonSet is used, each pod will receive
+	// flows from the node it is running on. When Deployment is used, the flows traffic received from nodes will
+	// be load-balanced. Note that in such a case, the number of replicas should be less or equal to the number of
+	// nodes, as extra-pods would be unused due to session affinity with the node IP.
```
Comment (OlivierCazade) [inline api/v1alpha1/flowcollector_types.go:198]: Nice I did not think of that limitation.

**Review by OlivierCazade — APPROVED** [review]
/lgtm

**Comment by jotak** [comment]
@nalhadef let me know if you would reformulate :)
The idea I want to express here, is that when the user configures flowlogs-pipeline as a Deployment, there's a limitation / known issue, in that it is worthless to have more replicas than the number of nodes, because those replicas would never be used

**Review by memodi — APPROVED** [review]
thanks! didn't know this.

**Pull request merged**